actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -480,6 +487,8 @@ def []=(name, options)
           options = { value: options }
         end
 
+        preprocess_options(options)
```

Comment:
The tough part is that we need `preprocess_options` to happen before `commit(options)` (so `options[:expires]` can be passed to `:expires_at`. When `@parent_jar[name] = options` comes along we'll hit `preprocess_options` again, yes.

Perhaps there's two things going on here that happen to look similar.

E.g. there's the duration support in `:expires`, but then there's the conversion that `Messages::Metadata` handles.

We could stash the `from_now` thing in `handle_options` and then have a separate method to handle the chained cookie jar cases.

```ruby
def expiry_options
  if options[:expires].is_a?(ActiveSupport::Duration)
    { expires_in: options[:expires] }
  else
    { expires_at: options[:expires] }
  end
end
```

That would spare us the complex delegation up the chained jars but cost us a type check. We could perhaps use `respond_to?(:from_now)` in both cases.